lib/uplinkc/uplink.go

```diff
@@ -23,8 +23,8 @@ type Uplink struct {
 // an error in cerr, when there is one.
 //
 // Caller must call close_uplink to close associated resources.
-func new_uplink(cfg C.UplinkConfig, cerr **C.char) C.UplinkRef {
-	scope := rootScope("") // TODO: pass in as argument
+func new_uplink(cfg C.UplinkConfig, tempDir *C.char, cerr **C.char) C.UplinkRef {
```

Comment:
I'm just wondering if we should make `tempDir` part of `UplinkConfig` to avoid keeping it as a separate parameter.

Reply:
Sounds reasonable

Reply:
My hesitation with that is that besides breaking symmetry with the go struct, the uplink config  is still not well defined as it currently contains a single `Volatile` struct. Additionally, keeping it as a parameter makes it more visible and clear that it's required (which apparently it is).